From d360f829d2fad80db0d206e2120ea0e5f986bc57 Mon Sep 17 00:00:00 2001
From: Yongjun Zhang <yzhang@cloudera.com>
Date: Fri, 6 May 2016 21:50:09 -0700
Subject: [PATCH 1377/1380] YARN-5048. DelegationTokenRenewer#skipTokenRenewal
 may throw NPE (Jian He via Yongjun Zhang)

(cherry picked from commit 47c41e7ac7e6b905a58550f8899f629c1cf8b138)
(cherry picked from commit 5ae5b869aa1be5b878957a5ec288c3dbc28900c3)

Change-Id: I5a1ef4f7e90c2b5f3c90630b51c8ae226a3acdda
(cherry picked from commit ef81ea1abc8c1476673314d98211bf5dc03ad871)
(cherry picked from commit 8d254164ac72cc0d02d4ebcfa36573d67a6c5420)
---
 .../security/DelegationTokenRenewer.java           |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/security/DelegationTokenRenewer.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/security/DelegationTokenRenewer.java
index 513e4d6..0412d1c 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/security/DelegationTokenRenewer.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/security/DelegationTokenRenewer.java
@@ -542,9 +542,14 @@ public boolean cancel() {
    */
   private boolean skipTokenRenewal(Token<?> token)
       throws IOException {
+
     @SuppressWarnings("unchecked")
-    Text renewer = ((Token<AbstractDelegationTokenIdentifier>)token).
-        decodeIdentifier().getRenewer();
+    AbstractDelegationTokenIdentifier identifier =
+        ((Token<AbstractDelegationTokenIdentifier>) token).decodeIdentifier();
+    if (identifier == null) {
+      return false;
+    }
+    Text renewer = identifier.getRenewer();
     return (renewer != null && renewer.toString().equals(""));
   }
 
-- 
1.7.9.5

